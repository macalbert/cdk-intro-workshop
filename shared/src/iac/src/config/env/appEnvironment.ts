export enum AppEnvironment {
    Production = "Production",
    Development = "Development",
    Staging = "Staging",
    GithubAction = "GithubAction",
    Test = "Test",
}
