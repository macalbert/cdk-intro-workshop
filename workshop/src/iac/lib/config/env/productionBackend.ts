import {
    ApiGatewayStack,
    type ApiGatewayStackProps,
    AppEnvironment,
    ApiLambdaStack,
    type ApiLambdaStackProps,
    S3Stack,
    type S3StackProps,
    StackBuildPart,
    type StackBuildProps,
} from "@m47/shared-iac";
import { type Stack } from "aws-cdk-lib";
import { SubnetType } from "aws-cdk-lib/aws-ec2";
import type { IFunction } from "aws-cdk-lib/aws-lambda";

export interface ProductionBackendProps extends StackBuildProps {
    clusterName: string;
    serviceName: string;
    exposedPort: number;
    sourceCodePath: string;
    dockerfileApi: string;
}

export class ProductionBackend extends StackBuildPart {
    props: ProductionBackendProps;

    environment = AppEnvironment.Production;

    constructor(buildProps: ProductionBackendProps) {
        super(buildProps);

        this.props = buildProps;
        this.props.githubRepo = this.formatRepoNameForCloudFormation();
    }

    build(): Stack[] {
        
        const apiLambdaStack = this.createApiLambdaStack(
            "ApiLambda",
            this.props.dockerfileApi,
            "api",
        );

        const apiGatewayStack = this.createApiGateway(
            apiLambdaStack.lambdaFunction,
            "ApiGateway",
            "api-cdk-workshop",
        );

        return [
            apiLambdaStack,
            apiGatewayStack,
        ];
    }

    private createApiLambdaStack(
        name: string,
        dockerfile: string,
        subdomain: string,
    ): ApiLambdaStack {
        const lambdaExportProps: ApiLambdaStackProps = {
            name: name,
            githubRepo: this.props.githubRepo,
            pathDockerFile: this.props.sourceCodePath,
            env: this.props.env,
            vpc: this.props.vpc,
            envName: AppEnvironment.Production,
            timeoutSeconds: 29,
            memorySizeMbs: 1024,
            vpcSubnets: {
                subnetType: SubnetType.PRIVATE_WITH_EGRESS,
            },
            stackName: `${this.props.githubRepo}-${name}`,
            functionName: `${this.props.githubRepo}-${subdomain}`.toLowerCase(),
            dockerFile: dockerfile,
        };

        const stack = new ApiLambdaStack(this.props.scope, lambdaExportProps);

        return stack;
    }

    private createApiGateway(
        lambdaFunction: IFunction,
        name: string,
        subdomain: string,
    ): ApiGatewayStack {
        const apiProps: ApiGatewayStackProps = {
            env: this.props.env,
            name: name,
            envName: AppEnvironment.Production,
            lambdaFunction: lambdaFunction,
            stackName: `${this.props.githubRepo}-${name}`,
            subdomain: subdomain,
            githubRepo: this.props.githubRepo,
            certificateArn: `arn:aws:acm:us-east-1:${this.props.env.account}:certificate/be63062d-5316-47af-9f94-819c1dc02853`,
            domain: "m47.io",
        };

        return new ApiGatewayStack(this.props.scope, apiProps);
    }
}
