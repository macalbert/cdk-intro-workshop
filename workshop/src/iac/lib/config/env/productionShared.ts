import {
    AppEnvironment,
    PipelineStack,
    type PipelineStackProps,
    RdsStack,
    type RdsStackProps,
    StackBuildPart,
    type StackBuildProps,
    WindowsBastionStack,
    type WindowsBastionStackProps,
} from "@m47/shared-iac";
import { RemovalPolicy, type Stack } from "aws-cdk-lib";
import { InstanceType, InstanceClass, InstanceSize } from "aws-cdk-lib/aws-ec2";
import { PostgresEngineVersion } from "aws-cdk-lib/aws-rds";

export class ProductionShared extends StackBuildPart {
    props: StackBuildProps;

    constructor(buildProps: StackBuildProps) {
        super(buildProps);

        this.props = buildProps;
    }

    build(): Stack[] {
        const pipelineStack = this.createPipelineStack();
        const rdsStack = this.createRdsStack();
        const createBastionStack = this.createBastionStack();

        return [
            rdsStack,
            pipelineStack,
            createBastionStack,
        ];
    }

    private createPipelineStack(): PipelineStack {
        const pipeline: PipelineStackProps = {
            name: "Pipeline",
            branch: this.props.branch,
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            secretTokenArn: `arn:aws:secretsmanager:${this.props.env.region}:${this.props.env.account}:secret:github-access-token-secret-dth2kY`,
            testBuildSpec: [
                "workshop/src/iac/buildspecs/test/iac.yml",
                "workshop/src/iac/buildspecs/test/backend.yml",
            ],
            deployBuildSpec: [
                "workshop/src/iac/buildspecs/production/backendApi.yml",
            ],
            testProjectName: `${this.props.githubRepo}-Tests`,
            vpc: this.props.vpc,
            env: this.props.env,
            bucketRemovalPolicy: RemovalPolicy.DESTROY,
            domain: "workshop.com",
            stackName: `${this.formatRepoNameForCloudFormation()}-CodePipeline`,
            manualApproval: false,
            shouldCreateNewSlackChannel: false,
        };

        return new PipelineStack(this.props.scope, pipeline);
    }

    private createRdsStack(): RdsStack {
        const props: RdsStackProps = {
            env: this.props.env,
            vpc: this.props.vpc,
            databaseName: "workshop",
            databaseUserName: "masteruser",
            backupRetentionDays: 14,
            storageSizeGb: 100,
            monitoringIntervalSeconds: 60,
            engineVersion: PostgresEngineVersion.VER_17_4,
            instanceType: InstanceType.of(InstanceClass.T3, InstanceSize.MICRO),
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            stackName: `${this.formatRepoNameForCloudFormation()}-rds`,
            name: "RdsPostgres",
        };

        return new RdsStack(this.props.scope, props);
    }

    private createBastionStack(): WindowsBastionStack {
        const windowsBastionStackProps: WindowsBastionStackProps = {
            vpc: this.props.vpc,
            githubRepo: this.props.githubRepo,
            envName: AppEnvironment.Production,
            name: "bastion",
            stackName: `${this.formatRepoNameForCloudFormation()}-Bastion`,
            env: this.props.env,
        };

        return new WindowsBastionStack(
            this.props.scope,
            windowsBastionStackProps,
        );
    }
}
