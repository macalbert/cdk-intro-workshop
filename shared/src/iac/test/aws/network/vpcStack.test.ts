import { Template } from "aws-cdk-lib/assertions";
import { SubnetType } from "aws-cdk-lib/aws-ec2";
import "source-map-support/register";
import { App, Stack } from "aws-cdk-lib";
import {
	type VPCStackProps,
	VpcStack,
} from "../../../src/aws/network/vpcStack";
import { AppEnvironment } from "../../../src/config/env/appEnvironment";

describe("VpcStackTest", () => {
	const expected: VPCStackProps = {
		name: "VpcStack",
		stackName: "shared",
		maxAzs: 2,
		natGateways: 2,
		enableDnsHostnames: true,
		enableDnsSupport: true,
		envName: AppEnvironment.Test,
		vpcName: "private-M47Vpc",
		vpcCdir: "22.0.0.0/16",
		subnetConfiguration: [
			{
				cidrMask: 20,
				name: "Public1",
				subnetType: SubnetType.PUBLIC,
			},
			{
				cidrMask: 20,
				name: "Private1",
				subnetType: SubnetType.PRIVATE_WITH_EGRESS,
			},
		],
		githubRepo: "shared-test",
	};
	const env = {
		account: "account",
		region: "eu-west-1",
	};

	test("Should_MatchSnapshotAndResources_When_VpcStackIsCreated", () => {
		// Arrange
		const stack = new Stack(new App(), "VpcStackTest", { env: env });

		// Act
		const vpcStack = new VpcStack(stack, expected);

		// Assert
		const template = Template.fromStack(vpcStack);
		expect(template.toJSON()).toMatchSnapshot("VpcStackTest");
		template.resourceCountIs("AWS::EC2::VPC", 1);
		template.hasResourceProperties("AWS::EC2::VPC", {
			EnableDnsSupport: expected.enableDnsSupport,
			EnableDnsHostnames: expected.enableDnsHostnames,
			CidrBlock: expected.vpcCdir,
		});
		template.resourceCountIs("AWS::EC2::InternetGateway", 1);
		template.resourceCountIs("AWS::EC2::VPCGatewayAttachment", 1);
		template.resourceCountIs("AWS::EC2::Subnet", 4);
	});
});
