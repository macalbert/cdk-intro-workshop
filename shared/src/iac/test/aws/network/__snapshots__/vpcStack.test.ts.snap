// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VpcStackTest Should_MatchSnapshotAndResources_When_VpcStackIsCreated: VpcStackTest 1`] = `
{
  "Outputs": {
    "cfnOutputVpcId": {
      "Description": "Created VPC ID",
      "Export": {
        "Name": "VpcStack:vpcId",
      },
      "Value": {
        "Ref": "vpcId937700B3",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "vpcId937700B3": {
      "Properties": {
        "CidrBlock": "22.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "private-M47Vpc-Test",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "vpcIdIGW108455B5": {
      "Properties": {
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "private-M47Vpc-Test",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "vpcIdPrivate1Subnet1DefaultRoute9D408128": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "vpcIdPublic1Subnet1NATGateway5EB16AC0",
        },
        "RouteTableId": {
          "Ref": "vpcIdPrivate1Subnet1RouteTable42D8C801",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcIdPrivate1Subnet1RouteTable42D8C801": {
      "Properties": {
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Private1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcIdPrivate1Subnet1RouteTableAssociation675EA1FB": {
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcIdPrivate1Subnet1RouteTable42D8C801",
        },
        "SubnetId": {
          "Ref": "vpcIdPrivate1Subnet1SubnetACE5346C",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcIdPrivate1Subnet1SubnetACE5346C": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "22.0.32.0/20",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private1",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Private1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcIdPrivate1Subnet2DefaultRoute8F3C7EA8": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "vpcIdPublic1Subnet2NATGateway6CB106F1",
        },
        "RouteTableId": {
          "Ref": "vpcIdPrivate1Subnet2RouteTable83AB5D28",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcIdPrivate1Subnet2RouteTable83AB5D28": {
      "Properties": {
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Private1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcIdPrivate1Subnet2RouteTableAssociation08CCB002": {
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcIdPrivate1Subnet2RouteTable83AB5D28",
        },
        "SubnetId": {
          "Ref": "vpcIdPrivate1Subnet2Subnet91487ED7",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcIdPrivate1Subnet2Subnet91487ED7": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "22.0.48.0/20",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private1",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Private1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcIdPublic1Subnet1DefaultRoute72228A8F": {
      "DependsOn": [
        "vpcIdVPCGW33ACD385",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "vpcIdIGW108455B5",
        },
        "RouteTableId": {
          "Ref": "vpcIdPublic1Subnet1RouteTable31C18ABF",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcIdPublic1Subnet1EIPDC296CEC": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcIdPublic1Subnet1NATGateway5EB16AC0": {
      "DependsOn": [
        "vpcIdPublic1Subnet1DefaultRoute72228A8F",
        "vpcIdPublic1Subnet1RouteTableAssociation055876F9",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "vpcIdPublic1Subnet1EIPDC296CEC",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "vpcIdPublic1Subnet1SubnetD21C4A28",
        },
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcIdPublic1Subnet1RouteTable31C18ABF": {
      "Properties": {
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcIdPublic1Subnet1RouteTableAssociation055876F9": {
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcIdPublic1Subnet1RouteTable31C18ABF",
        },
        "SubnetId": {
          "Ref": "vpcIdPublic1Subnet1SubnetD21C4A28",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcIdPublic1Subnet1SubnetD21C4A28": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "22.0.0.0/20",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public1",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet1",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcIdPublic1Subnet2DefaultRouteE84FDE21": {
      "DependsOn": [
        "vpcIdVPCGW33ACD385",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "vpcIdIGW108455B5",
        },
        "RouteTableId": {
          "Ref": "vpcIdPublic1Subnet2RouteTable6FCD1F9B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcIdPublic1Subnet2EIP15C5A5DD": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "vpcIdPublic1Subnet2NATGateway6CB106F1": {
      "DependsOn": [
        "vpcIdPublic1Subnet2DefaultRouteE84FDE21",
        "vpcIdPublic1Subnet2RouteTableAssociation126B887D",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "vpcIdPublic1Subnet2EIP15C5A5DD",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "vpcIdPublic1Subnet2SubnetA44E3D9B",
        },
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "vpcIdPublic1Subnet2RouteTable6FCD1F9B": {
      "Properties": {
        "Tags": [
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "vpcIdPublic1Subnet2RouteTableAssociation126B887D": {
      "Properties": {
        "RouteTableId": {
          "Ref": "vpcIdPublic1Subnet2RouteTable6FCD1F9B",
        },
        "SubnetId": {
          "Ref": "vpcIdPublic1Subnet2SubnetA44E3D9B",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "vpcIdPublic1Subnet2SubnetA44E3D9B": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "22.0.16.0/20",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public1",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Environment",
            "Value": "Test",
          },
          {
            "Key": "Name",
            "Value": "VpcStackTest/m47-shared-test-vpcstack-test-stack/vpcId/Public1Subnet2",
          },
          {
            "Key": "Project",
            "Value": "shared-test",
          },
          {
            "Key": "StackId",
            "Value": "m47-shared-test-vpcstack-test-stack",
          },
        ],
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "vpcIdVPCGW33ACD385": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "vpcIdIGW108455B5",
        },
        "VpcId": {
          "Ref": "vpcId937700B3",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
